source/fundamentals/connection/socks.txt

```diff
@@ -0,0 +1,126 @@
+       set the ``proxyHost`` option, the value of this option defaults
+       to ``1080``.
+
+   * - **proxyUsername**
+     - String
+     - Specifies the username for authentication to the SOCKS5
+       proxy server. The driver ignores ``null`` and empty string values for
+       this setting.
+
+   * - **proxyPassword**
+     - String
+     - Specifies the password for authentication to the SOCKS5
+       proxy server. The driver ignores ``null`` and empty string values
+       for this setting.
+
+
+Examples
+--------
+
+The following examples show how to instantiate a ``MongoClient`` that connects
+to MongoDB by using a SOCKS5 proxy by using a ``MongoClientSettings`` instance
+or a connection string. These examples use the placeholder values described in
+the :ref:`socks-proxy-settings` section. Replace the placeholders with your
+proxy settings.
+
+Specify Proxy Settings in the MongoClientSettings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following code example shows how to specify your SOCKS5 proxy settings by
+using the ``MongoClientSettings`` builder:
+
+.. code-block:: java
+   :emphasize-lines: 4-11
+
+     MongoClient mongoClient = MongoClients.create(
+         MongoClientSettings.builder()
+             .applyConnectionString(new ConnectionString("mongodb+srv://myDatabaseUser:myPassword@example.org/"))
+             .applyToSocketSettings(builder ->
+                     builder.applyToProxySettings(proxyBuilder ->
+                             proxyBuilder
+                                 .host("<proxyHost>")
+                                 .port(<proxyPort>)
+                                 .username("<proxyUsername>")
+                                 .password("<proxyPassword>")
+                             )
+             ).build());
+
+Specify Proxy Settings in the Connection String
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This following code example shows how to specify your SOCKS5 proxy settings in
+your connection string:
+
+.. code-block:: java
+   :emphasize-lines: 2-5
+
+   String connectionString = "mongodb+srv://myDatabaseUser:myPassword@example.org/" +
+           "?proxyHost=<proxyHost>" +
+           "&proxyPort=<proxyPort>" +
+           "&proxyUsername=<proxyUsername>" +
+           "&proxyPassword=<proxyPassword>";
+
+   MongoClient mongoClient = MongoClients.create(connectionString);
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about the methods and types discussed in this guide, see the
+following API documentation:
+
+- `MongoClientSettings.Builder <{+api+}/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html>`__
+- `SocketSettings.Builder <{+api+}/mongodb-driver-core/com/mongodb/connection/SocketSettings.Builder.html>`__
+- `MongoClients.create() <{+api+}/mongodb-driver-sync/com/mongodb/client/MongoClients.html#create(com.mongodb.MongoClientSettings)>`__
+
+.. TODO
+   - provide the link for the proxysettings builder once the API docs are generated
```

Comment:
This will be replaced by the ProxySettings Builder API docs link once v4.11 is published.